source/common/stats/symbol_table_impl.h

```diff
@@ -19,36 +23,168 @@
 namespace Envoy {
 namespace Stats {
 
+/** A Symbol represents a string-token with a small index. */
 using Symbol = uint32_t;
+
+/** Efficient byte-encoded storage an array of tokens, which are typically < 127 */
+using SymbolStorage = uint8_t[];
+
+/** Transient representations of a vector of 32-bit symbols */
 using SymbolVec = std::vector<Symbol>;
 
+/** Forward declaration for the StatName class (see below) */
+class StatName;
+
 /**
- * Underlying SymbolTableImpl implementation which manages per-symbol reference counting.
+ * Represents an 8-bit encoding of a vector of symbols, used as a transient
+ * representation during encoding and prior to retained allocation.
+ */
+class SymbolEncoding {
+public:
+  /**
+   * Before destructing SymbolEncoding, you must call moveToStorage. This
+   * transfers ownership, and in particular, the responsibility to call
+   * SymbolTable::clear() on all referenced symbols. If we ever wanted
+   * to be able to destruct a SymbolEncoding without transferring it
+   * we could add a clear(SymbolTable&) method.
+   */
+  ~SymbolEncoding();
+
+  /**
+   * Encodes a token into the vec.
+   *
+   * @param symbol the symbol to encode.
+   */
+  void addSymbol(Symbol symbol);
+
+  /**
+   * Decodes a symbol from a uint8_t array. When the last symbol is found,
+   * false is returned as well as the symbol.
```

Comment:
False?

Reply:
done -- this was previously going one Symbol at a time and then I decided it was a much simpler interface to return the vector, and was more efficient too even if the vector is only used as a temp, because it leaves the SymbolTable locks held for less time.